rewrite-gradle/src/main/resources/RewriteGradleProject.groovy

```diff
@@ -15,150 +15,27 @@
-import org.gradle.jvm.tasks.*
-import org.gradle.jvm.toolchain.*
-import org.gradle.language.*
-import org.gradle.language.assembler.*
-import org.gradle.language.assembler.plugins.*
-import org.gradle.language.assembler.tasks.*
-import org.gradle.language.base.*
-import org.gradle.language.base.artifact.*
-import org.gradle.language.base.compile.*
-import org.gradle.language.base.plugins.*
-import org.gradle.language.base.sources.*
-import org.gradle.language.java.artifact.*
-import org.gradle.language.jvm.tasks.*
-import org.gradle.language.plugins.*
-import org.gradle.language.rc.*
-import org.gradle.language.rc.plugins.*
-import org.gradle.language.rc.tasks.*
-import org.gradle.maven.*
-import org.gradle.model.*
-import org.gradle.normalization.*
-import org.gradle.platform.base.*
-import org.gradle.platform.base.binary.*
-import org.gradle.platform.base.component.*
-import org.gradle.platform.base.plugins.*
-import org.gradle.plugin.devel.*
-import org.gradle.plugin.devel.plugins.*
-import org.gradle.plugin.devel.tasks.*
-import org.gradle.plugin.management.*
-import org.gradle.plugin.use.*
-import org.gradle.plugins.ear.*
-import org.gradle.plugins.ear.descriptor.*
-import org.gradle.plugins.ide.*
-import org.gradle.plugins.ide.api.*
-import org.gradle.plugins.ide.eclipse.*
-import org.gradle.plugins.ide.idea.*
-import org.gradle.plugins.signing.*
-import org.gradle.plugins.signing.signatory.*
-import org.gradle.plugins.signing.signatory.pgp.*
-import org.gradle.plugins.signing.type.*
-import org.gradle.plugins.signing.type.pgp.*
-import org.gradle.process.*
-import org.gradle.testing.base.*
-import org.gradle.testing.base.plugins.*
-import org.gradle.testing.jacoco.plugins.*
-import org.gradle.testing.jacoco.tasks.*
-import org.gradle.testing.jacoco.tasks.rules.*
-import org.gradle.testkit.runner.*
-import org.gradle.util.*
-import org.gradle.vcs.*
-import org.gradle.vcs.git.*
-import org.gradle.work.*
-import org.gradle.workers.*
+import org.gradle.api.Action
+import org.gradle.api.JavaVersion
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.artifacts.Dependency
+import org.gradle.api.artifacts.ModuleDependency
+import org.gradle.api.artifacts.dsl.DependencyHandler
+import org.gradle.api.artifacts.dsl.RepositoryHandler
+import org.gradle.api.artifacts.repositories.IvyArtifactRepository
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository
+import org.gradle.api.file.FileCollection
+import org.gradle.api.file.FileTree
+import org.gradle.api.file.FileTreeElement
+import org.gradle.api.specs.Spec
+import org.gradle.api.tasks.testing.TestFilter
+import org.gradle.api.tasks.testing.TestFrameworkOptions
+import org.gradle.api.tasks.testing.junit.JUnitOptions
+import org.gradle.api.tasks.testing.junitplatform.JUnitPlatformOptions
+import org.gradle.api.tasks.testing.testng.TestNGOptions
+import org.gradle.process.JavaForkOptions
+import org.gradle.process.ProcessForkOptions
```

Comment:
Because `RewriteGradleProject.groovy` is now being used as a `@groovy.transform.BaseScript` class, this means that having the totality of imports is no longer necessary.

Additional observation while investigating the LSTs/ASTs is that even though these imports were present, types that we would have expected to be resolved were actually just being parsed as `J.Identifier` types by the Groovy compiler anyway.

ex:
```
tasks.register("test", Test) { // Test I would have thought would be a class reference, but it's actually just an identifier in the groovy AST
}
```

Reply:
At least some of these attribution failures are due to failing to load the class. I looked into `Test` specifically at one point and it fails because some of the types used in its implementation aren't present. It's frustrating because we only care about its public API, but the classloader fails due to us not including types that are used in its implementation. Would be resolved if we just stuffed the _entire_ gradle distribution in there, which is maybe what we have to do